install/migrations/update_10.0.11_to_10.0.12/duplicate_searchopt_fixes.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ * @var array $CFG_GLPI
+ * @var \Migration $migration
+ */
+
+if ($CFG_GLPI["lock_use_lock_item"] ?? false) {
```

Comment:
I'm not sure this is a good idea to rely on CFG_GLPI. It may change in the future, and it may be affected by plugins also.
I think we must here hardcode known core itemtypes that must be migrated only.

Reply:
The main problem here is that these numbers are related to `Domain` search options when the itemtype is not present in `lock_use_lock_item`, otherwise they are related to `ObjectLock`.

Anyway, I agree we should not rely on `$CFG_GLPI` here. Migrations should rely only on database values, to limit the probability to break them whenever we want to change the internal GLPI API/global variables management. We should probably use the `$DB->request()` method to fetch this configuration value.